lib/object/errors/InvalidDPObjectStateError.js

```diff
@@ -0,0 +1,47 @@
+class InvalidDPObjectStateError extends Error {
```

Comment:
It looks like overcomplication. What if we just have one error `DataIsNotAllowedWithActionDeleteError` or something like that?